commands/list_endpoints.go

```diff
@@ -61,30 +61,59 @@ func endpointsTable(args listEndpointsArguments) string {
 			color.YellowString("gsctl login <email> -e <endpoint>"))
 	}
 
-	// table headers
-	output := []string{
-		strings.Join([]string{
-			color.CyanString("ENDPOINT URL"),
-			color.CyanString("EMAIL"),
-			color.CyanString("SELECTED"),
-			color.CyanString("LOGGED IN"),
-		}, "|"),
-	}
-
 	// get keys (URLs) and sort by them
 	endpointURLs := make([]string, 0, len(config.Config.Endpoints))
 	for u := range config.Config.Endpoints {
 		endpointURLs = append(endpointURLs, u)
 	}
 
+	// detect if we want to show the alias column
+	haveAlias := false
+	for _, endpoint := range endpointURLs {
+		if config.Config.Endpoints[endpoint].Alias != "" {
```

Comment:
Accessing the map item with `config.Config.Endpoints[endpoint]` is done quite often. This could be improved. It is also that this code can panic in case the map does not hold what you need. Even if you validate this above, this quite big function relies on the validation in a lot of places and this can go wrong quite quickly based on my personal experiences.